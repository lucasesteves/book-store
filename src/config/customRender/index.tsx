import React from "react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme from "~/styles/theme";
import { store } from "~/store";

export default (component: React.ReactElement, options?: RenderOptions) => {
  const Wrapper: any = ({ children }: { children: React.ReactChildren }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme()}>
        <MemoryRouter>{children}</MemoryRouter>
      </ThemeProvider>
    </Provider>
  );

  return render(component, { wrapper: Wrapper, ...options });
};
