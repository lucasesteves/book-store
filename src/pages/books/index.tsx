import React, { useState } from "react";
import * as S from "./styles";
import { Loading } from "~/components";

const mock = [
  {
    title: "Titulo1",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas cursu",
  },
];

const Books: React.FC = () => {
  const [load, setLoad] = useState(false);
  return (
    <S.Wrapper>
      {[1, 2, 3, 4, 5, 6, 7, 8].map((item) => (
        <S.Card
          key={item}
          title={mock[0].title}
          description={mock[0].description}
        >
          <S.Button onClick={() => setLoad(true)}>
            {load ? <Loading /> : "Rent"}
          </S.Button>
        </S.Card>
      ))}
    </S.Wrapper>
  );
};

export default Books;
