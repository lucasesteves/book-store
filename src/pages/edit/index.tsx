import React, { useState } from "react";
import * as S from "./styles";
import { Modal } from "~/components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus } from "@fortawesome/free-solid-svg-icons";

const mock = [
  {
    title: "Titulo1",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas cursu",
  },
];

const Create: React.FC = () => {
  const [modal, setModal] = useState({
    add: false,
    edit: false,
    remove: false,
  });
  const [form, setForm] = useState({ title: "", description: "" });

  return (
    <S.Wrapper>
      {[1, 2, 3, 4, 5, 6, 7, 8].map((item) => (
        <S.Card
          key={item}
          title={mock[0].title}
          description={mock[0].description}
        >
          <S.Button
            color="info"
            onClick={() => setModal({ ...modal, edit: true })}
          >
            Edit
          </S.Button>
          <S.Button
            variant="outline"
            color="error"
            onClick={() => setModal({ ...modal, remove: true })}
          >
            Delete
          </S.Button>
        </S.Card>
      ))}

      <S.AddButton onClick={() => setModal({ ...modal, add: true })}>
        <FontAwesomeIcon icon={faPlus} />
      </S.AddButton>

      <Modal
        open={modal.add || modal.edit}
        title={modal.add ? "Add Book" : "Edit Book"}
        onClose={() => setModal({ ...modal, add: false, edit: false })}
      >
        <S.Input
          type="text"
          placeholder="Title"
          value={form.title}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
            setForm({ ...form, title: event.target.value })
          }
        />

        <S.Input
          type="text"
          placeholder="Description"
          value={form.description}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
            setForm({ ...form, description: event.target.value })
          }
        />

        <S.Button
          onClick={() => setModal({ ...modal, add: false, edit: false })}
        >
          Save
        </S.Button>
      </Modal>

      <Modal
        open={modal.remove}
        title="Remove book"
        onClose={() => setModal({ ...modal, remove: false })}
      >
        <S.ErrorMessage>Do you really want delete this book?</S.ErrorMessage>
        <S.ErrorContent>
          <S.Button
            variant="outline"
            color="info"
            onClick={() => setModal({ ...modal, remove: false })}
          >
            Cancel
          </S.Button>
          <S.Button
            color="error"
            onClick={() => setModal({ ...modal, remove: false })}
          >
            Confirm
          </S.Button>
        </S.ErrorContent>
      </Modal>
    </S.Wrapper>
  );
};

export default Create;
