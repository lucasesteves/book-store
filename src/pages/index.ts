export { default as Login } from "./login";
export { default as Books } from "./books";
export { default as Edit } from "./edit";
export { default as Detail } from "./detail";
export { default as RentedBooks } from "./rentedBooks";
export { default as Template } from "./template";
