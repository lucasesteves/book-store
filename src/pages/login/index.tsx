import React, { useState } from "react";
import * as S from "./styles";
import { useDispatch, useSelector } from "react-redux";
import { loadingRequest, loginRequest } from "~/store/login/actions";
import { ApplicationState } from "~/store";
import { Loading } from "~/components";
import { timeout } from "~/utils";

const Login: React.FC = () => {
  const dispatch = useDispatch();
  const [name, setName] = useState("");
  const { loading } = useSelector(({ login }: ApplicationState) => login);

  return (
    <S.Wrapper>
      <S.Card>
        <S.Title color="primary" align="center">
          Book Store
        </S.Title>
        <S.Input
          type="text"
          placeholder="enter with your name"
          value={name}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
            setName(event.target.value)
          }
        />
        <S.Button
          disabled={!name}
          onClick={() => {
            dispatch(loadingRequest());
            timeout(() => {
              dispatch(loginRequest(name));
            });
          }}
        >
          {loading ? <Loading /> : "Entrar"}
        </S.Button>
      </S.Card>
    </S.Wrapper>
  );
};

export default Login;
