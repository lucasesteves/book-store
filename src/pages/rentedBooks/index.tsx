import React, { useState, useEffect } from "react";
import * as S from "./styles";
import { useDispatch, useSelector } from "react-redux";
import { ApplicationState } from "~/store";
import { removeRentBook, loadingRequest } from "~/store/dashboard/actions";
import { Loading, Title } from "~/components";
import { MOCK_DATA } from "~/store/dashboard/types";
import { DEFAULT_ITEM, timeout } from "~/utils";

const RentedBooks: React.FC = () => {
  const dispatch = useDispatch();
  const [select, setSelect] = useState<MOCK_DATA>(DEFAULT_ITEM);
  const [list, setList] = useState<MOCK_DATA[]>([]);
  const { loading, rentedBooks, search } = useSelector(
    ({ dashboard }: ApplicationState) => dashboard
  );

  useEffect(() => {
    const searchFilter = rentedBooks.filter(
      (item) => item.title.toLowerCase().indexOf(search) > -1
    );
    setList(searchFilter);
  }, [search]);

  useEffect(() => {
    setList(rentedBooks);
  }, [rentedBooks]);

  return (
    <S.Wrapper>
      {rentedBooks.length > 0 ? (
        list.map((item) => (
          <S.Card
            key={item.id}
            title={item.title}
            description={item.description}
          >
            <S.Button
              data-testid="button-back"
              onClick={() => {
                setSelect(item);
                dispatch(loadingRequest());
                timeout(() => {
                  dispatch(removeRentBook({ ...item, status: "avaliable" }));
                });
              }}
            >
              {loading && select.id === item.id ? <Loading /> : "Give back"}
            </S.Button>
          </S.Card>
        ))
      ) : (
        <Title>You donÂ´t have books to show</Title>
      )}
    </S.Wrapper>
  );
};

export default RentedBooks;
