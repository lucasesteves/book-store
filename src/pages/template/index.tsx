import React, { useState } from "react";
import * as S from "./styles";
import * as T from "./types";
import { Drawer } from "~/components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars } from "@fortawesome/free-solid-svg-icons";
import { searchBook } from "~/store/dashboard/actions";
import { useDispatch, useSelector } from "react-redux";
import { ApplicationState } from "~/store";

const Template: React.FC<T.TemplateProps> = ({ children }: T.TemplateProps) => {
  const dispatch = useDispatch();
  const [open, setOpen] = useState(false);
  const { search } = useSelector(
    ({ dashboard }: ApplicationState) => dashboard
  );

  return (
    <>
      <S.Header>
        <S.Input
          type="text"
          placeholder="search book"
          value={search}
          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
            dispatch(searchBook(event.target.value.toLowerCase()));
          }}
        />
        <S.Icon onClick={() => setOpen(true)}>
          <FontAwesomeIcon icon={faBars} />
        </S.Icon>
      </S.Header>

      <S.MobileDrawer open={open}>
        <Drawer size="large" onClose={() => setOpen(false)} />
      </S.MobileDrawer>

      <S.Wrapper>
        <S.DrawerContainer>
          <Drawer />
        </S.DrawerContainer>
        <S.Container>{children}</S.Container>
      </S.Wrapper>
    </>
  );
};

export default Template;
