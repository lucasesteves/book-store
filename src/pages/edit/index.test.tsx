import React from "react";
import "@testing-library/jest-dom";
import { Edit } from "~/pages";
import customRender from "~/config/customRender";

describe("Edit page", () => {
  it("Should render correctly", () => {
    expect(customRender(<Edit />)).toMatchSnapshot();
  });
});
