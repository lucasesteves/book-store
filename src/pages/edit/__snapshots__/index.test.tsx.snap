// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit page Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ikJyIC fPfBUR"
      >
        <div
          class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
        >
          <img
            alt="Harry Potter 1"
            class="sc-crHmcD eWcVlw"
            src="book.jpg"
          />
          <span
            class="sc-egiyK imxGsE"
          >
            Harry Potter 1
          </span>
          <span
            class="sc-bqiRlB LDmGk"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
          </span>
          <div
            class="sc-ksdxgE eKpHrr"
          >
            <button
              class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
              color="info"
            >
              Edit
            </button>
            <button
              class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
              color="error"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
        >
          <img
            alt="Harry Potter 2"
            class="sc-crHmcD eWcVlw"
            src="book.jpg"
          />
          <span
            class="sc-egiyK imxGsE"
          >
            Harry Potter 2
          </span>
          <span
            class="sc-bqiRlB LDmGk"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
          </span>
          <div
            class="sc-ksdxgE eKpHrr"
          >
            <button
              class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
              color="info"
            >
              Edit
            </button>
            <button
              class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
              color="error"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
        >
          <img
            alt="Harry Potter 3"
            class="sc-crHmcD eWcVlw"
            src="book.jpg"
          />
          <span
            class="sc-egiyK imxGsE"
          >
            Harry Potter 3
          </span>
          <span
            class="sc-bqiRlB LDmGk"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
          </span>
          <div
            class="sc-ksdxgE eKpHrr"
          >
            <button
              class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
              color="info"
            >
              Edit
            </button>
            <button
              class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
              color="error"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
        >
          <img
            alt="Harry Potter 4"
            class="sc-crHmcD eWcVlw"
            src="book.jpg"
          />
          <span
            class="sc-egiyK imxGsE"
          >
            Harry Potter 4
          </span>
          <span
            class="sc-bqiRlB LDmGk"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
          </span>
          <div
            class="sc-ksdxgE eKpHrr"
          >
            <button
              class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
              color="info"
            >
              Edit
            </button>
            <button
              class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
              color="error"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
        >
          <img
            alt="Harry Potter 5"
            class="sc-crHmcD eWcVlw"
            src="book.jpg"
          />
          <span
            class="sc-egiyK imxGsE"
          >
            Harry Potter 5
          </span>
          <span
            class="sc-bqiRlB LDmGk"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
          </span>
          <div
            class="sc-ksdxgE eKpHrr"
          >
            <button
              class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
              color="info"
            >
              Edit
            </button>
            <button
              class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
              color="error"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          class="sc-gWXbKe ceVdfT"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus fa-w-14 "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="sc-hBUSln dvkdBf"
        >
          <div
            class="sc-fotOHu WEMGP"
          >
            <div
              class="sc-fFeiMQ bmpUhW"
            >
              <span
                class="sc-bkkeKt flbXbU"
              >
                Edit Book
              </span>
              <div
                class="sc-ieecCq kgOxaW"
                data-testid="button-close"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-times fa-w-11 "
                  data-icon="times"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 352 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <input
              class="sc-gsDKAQ cArwkB sc-cCcXHH fmtylA"
              color="primary"
              placeholder="Title"
              type="text"
              value=""
            />
            <textarea
              class="sc-hGPBjI eEuEjr sc-cidDSM hbOZnf"
              color="primary"
              placeholder="Description"
              rows="5"
            />
            <button
              class="sc-bdvvtL dPUGtn sc-hiCibw ikIwjb"
              disabled=""
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="sc-hBUSln dvkdBf"
        >
          <div
            class="sc-fotOHu WEMGP"
          >
            <div
              class="sc-fFeiMQ bmpUhW"
            >
              <span
                class="sc-bkkeKt flbXbU"
              >
                Remove book
              </span>
              <div
                class="sc-ieecCq kgOxaW"
                data-testid="button-close"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-times fa-w-11 "
                  data-icon="times"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 352 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <span
              class="sc-jcFjpl bbaxey"
            >
              Do you really want delete this book?
            </span>
            <div
              class="sc-caiLqq kZrdHB"
            >
              <button
                class="sc-bdvvtL kKMiCl sc-hiCibw ikIwjb"
                color="info"
              >
                Cancel
              </button>
              <button
                class="sc-bdvvtL fCbDpJ sc-hiCibw ikIwjb"
                color="error"
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ikJyIC fPfBUR"
    >
      <div
        class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
      >
        <img
          alt="Harry Potter 1"
          class="sc-crHmcD eWcVlw"
          src="book.jpg"
        />
        <span
          class="sc-egiyK imxGsE"
        >
          Harry Potter 1
        </span>
        <span
          class="sc-bqiRlB LDmGk"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
        </span>
        <div
          class="sc-ksdxgE eKpHrr"
        >
          <button
            class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
            color="info"
          >
            Edit
          </button>
          <button
            class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
            color="error"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
      >
        <img
          alt="Harry Potter 2"
          class="sc-crHmcD eWcVlw"
          src="book.jpg"
        />
        <span
          class="sc-egiyK imxGsE"
        >
          Harry Potter 2
        </span>
        <span
          class="sc-bqiRlB LDmGk"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
        </span>
        <div
          class="sc-ksdxgE eKpHrr"
        >
          <button
            class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
            color="info"
          >
            Edit
          </button>
          <button
            class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
            color="error"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
      >
        <img
          alt="Harry Potter 3"
          class="sc-crHmcD eWcVlw"
          src="book.jpg"
        />
        <span
          class="sc-egiyK imxGsE"
        >
          Harry Potter 3
        </span>
        <span
          class="sc-bqiRlB LDmGk"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
        </span>
        <div
          class="sc-ksdxgE eKpHrr"
        >
          <button
            class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
            color="info"
          >
            Edit
          </button>
          <button
            class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
            color="error"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
      >
        <img
          alt="Harry Potter 4"
          class="sc-crHmcD eWcVlw"
          src="book.jpg"
        />
        <span
          class="sc-egiyK imxGsE"
        >
          Harry Potter 4
        </span>
        <span
          class="sc-bqiRlB LDmGk"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
        </span>
        <div
          class="sc-ksdxgE eKpHrr"
        >
          <button
            class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
            color="info"
          >
            Edit
          </button>
          <button
            class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
            color="error"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="sc-iqseJM izWPfs sc-jJoQJp hbhRpG"
      >
        <img
          alt="Harry Potter 5"
          class="sc-crHmcD eWcVlw"
          src="book.jpg"
        />
        <span
          class="sc-egiyK imxGsE"
        >
          Harry Potter 5
        </span>
        <span
          class="sc-bqiRlB LDmGk"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel finibus dolor. Sed quis faucibus lectus. Donec feugiat nisi cursus velit vestibulum euismod. Praesent lacus metus, cursus tristique est non, porttitor volutpat urna. Praesent ac nisi gravida, auctor libero quis, imperdiet risus. In blandit dignissim vehicula. Ut condimentum dictum arcu in scelerisque.
        </span>
        <div
          class="sc-ksdxgE eKpHrr"
        >
          <button
            class="sc-bdvvtL FgMTE sc-hiCibw ikIwjb"
            color="info"
          >
            Edit
          </button>
          <button
            class="sc-bdvvtL bwZWSl sc-hiCibw ikIwjb"
            color="error"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="sc-gWXbKe ceVdfT"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="sc-hBUSln dvkdBf"
      >
        <div
          class="sc-fotOHu WEMGP"
        >
          <div
            class="sc-fFeiMQ bmpUhW"
          >
            <span
              class="sc-bkkeKt flbXbU"
            >
              Edit Book
            </span>
            <div
              class="sc-ieecCq kgOxaW"
              data-testid="button-close"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <input
            class="sc-gsDKAQ cArwkB sc-cCcXHH fmtylA"
            color="primary"
            placeholder="Title"
            type="text"
            value=""
          />
          <textarea
            class="sc-hGPBjI eEuEjr sc-cidDSM hbOZnf"
            color="primary"
            placeholder="Description"
            rows="5"
          />
          <button
            class="sc-bdvvtL dPUGtn sc-hiCibw ikIwjb"
            disabled=""
          >
            Edit
          </button>
        </div>
      </div>
      <div
        class="sc-hBUSln dvkdBf"
      >
        <div
          class="sc-fotOHu WEMGP"
        >
          <div
            class="sc-fFeiMQ bmpUhW"
          >
            <span
              class="sc-bkkeKt flbXbU"
            >
              Remove book
            </span>
            <div
              class="sc-ieecCq kgOxaW"
              data-testid="button-close"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <span
            class="sc-jcFjpl bbaxey"
          >
            Do you really want delete this book?
          </span>
          <div
            class="sc-caiLqq kZrdHB"
          >
            <button
              class="sc-bdvvtL kKMiCl sc-hiCibw ikIwjb"
              color="info"
            >
              Cancel
            </button>
            <button
              class="sc-bdvvtL fCbDpJ sc-hiCibw ikIwjb"
              color="error"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
